funsor/einsum/numpy_map.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2017-2019 Uber Technologies, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import operator
+
+from functools import reduce
+
+import funsor.ops as ops
+from funsor.einsum.util import Tensordot, broadcast_all, rename_equation
+
+
+def einsum(equation, *operands):
+    """
+    Forward-max-sum backward-argmax implementation of einsum.
+    This assumes all operands have a ``._pyro_dims`` attribute set.
+    """
+    # TODO: the implementation skips "backward" logic
```

Comment:
The "backward" logic will be handled by `funsor.adjoint`, so you can remove this TODO.